test/test_torch.py

```diff
@@ -1142,6 +1142,36 @@ def test_pdist_norm(self):
                         self.assertEqual(expected.shape, actual.shape)
                         self.assertTrue(torch.allclose(expected, actual))
 
+
+    def test_cdist_empty(self):
+        devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
+        for device in devices:
+            for r1 in [0, 2]:
+                for m in[0, 5]:
+                    for r2 in[0, 3]:
+                        x = torch.zeros((r1, m), device=device)
+                        y = torch.zeros((r2, m), device=device)
+                        if r1 == 0 or r2 == 0:
+                            self.assertEqual(torch.empty(r1, r2, device=device), torch.cdist(x, y))
+                        else:
+                            self.assertEqual(torch.zeros(r1, r2, device=device), torch.cdist(x, y))
```

Comment:
this isn't testing anything we particularly care about here -- instead of choosing individual dimension sizes, how about selecting entire shapes and not checking when the shapes aren't empty?

Reply:
Fixed